packages/server/graphql/public/mutations/createOAuthAPIProvider.ts

```diff
@@ -0,0 +1,62 @@
+import {SubscriptionChannel} from 'parabol-client/types/constEnums'
+import {generateOAuthClientId, generateOAuthClientSecret} from '../../../oauth2/credentials'
+import getKysely from '../../../postgres/getKysely'
+import {selectOAuthAPIProvider} from '../../../postgres/select'
+import publish from '../../../utils/publish'
+import type {MutationResolvers} from '../resolverTypes'
+
+export const createOAuthAPIProvider: MutationResolvers['createOAuthAPIProvider'] = async (
+  _,
+  args,
+  _context
+) => {
+  const {input} = args
+  let {scopes = []} = input
+  const {orgId, name, redirectUris} = input
+
+  const clientId = generateOAuthClientId()
+  const clientSecret = generateOAuthClientSecret()
+
+  if (scopes.length === 0) {
+    scopes = ['graphql:query', 'graphql:mutation']
+  }
+
+  const pg = getKysely()
+
+  const dbRow = {
+    orgId,
+    name,
+    clientId,
+    clientSecret,
+    redirectUris,
+    scopes
+  }
+
+  const {id: providerId} = await pg
+    .insertInto('OAuthAPIProvider')
+    .values(dbRow)
+    .returning('id')
+    .executeTakeFirstOrThrow()
+
+  const provider = await selectOAuthAPIProvider()
+    .where('id', '=', providerId)
+    .executeTakeFirstOrThrow()
+
+  const data = {
+    provider,
+    clientId: provider.clientId,
+    organization: {
+      id: orgId
+    }
+  }
+
+  publish(SubscriptionChannel.ORGANIZATION, orgId, 'CreateOAuthAPIProviderSuccess', data, {})
```

Comment:
-1 remove this call as well as `CreateOAuthAPIProviderSuccess` :
Having the mutation return `CreateOAuthAPIProviderPayload` and the subscription return `CreateOAuthAPIProviderSuccess` is a little confusing, we try to have the subscription & mutation payloads match so a mutation handler can be reused by both. Looking closer, we're publishing a provider & an organization object that only has the `id`, so it doesn't fulfill its contract.
The client is subscription to it, but isn't doing anything with the payload, so the whole thing is a noop, unless I'm missing something?